### ISSUE TITLE
Reconnecting storage to refactored code

### ISSUE THREAD

**Issue by kalicki1** [privacy-tech-lab/optmeowt-browser-extension#178]
This should be a smaller issue since it's been in progress for a while. I will keep working on this in issue #166 as a sub-issue. 

I am trying to reconnect the storage as it is implemented with local storage so that we only interface with the storage via a wrapper everywhere in the extension. I am currently trying to resolve some synchronous/asynchronouos issues related to this, for some reason by local storage calls don't quite get synced up before I pull the info back down during tests. Once this is resolved, I will push my changes for @kbeliauski and @stanleymarkman for their respective indexDB / frontend issues. 

**Side note:** changes made in relation to this issue have been uploaded to branch issue-166.

**Comment by kalicki1** [comment]
I just pushed some changes to branch #166 and fixed some of the links to `domainlist.js` and `storage.js`. The few functions that are working can now be used asynchronously with `await`. I have a few random tests floating in `background.js` for the time being. Overall, I think this will help clean up a lot of our code. 

Technically I implemented this by wrapping all of the storage calls with promises. 

@kbeliauski @stanleymarkman, its not fully implemented but maybe you can start looking when you have some time to get familiar with how it works for when my changes are done. I will fix up the rest in the coming days.

**Comment by kalicki1** [comment]
I managed to finish off the remainder of the functions connected to this issue and push cleaned up code to branch issue-166. The hierarchy of the code pertaining to this issue is as follows:

`background.js` <- `domainlist.js` <- `storage.js`, which `background.js` implemented each successive, abstracted file.

I will close this issue for now since this is resolved. If we need to fix the code, we can open it back up. @kbeliauski will continue to implement indexedDB in issue #170 using the wrappers I created here.
